extensions/wikia/SpecialMultipleLookup/MultiLookupHooks.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+class MultiLookupHooks {
+	static public function onContributionsToolLinks( $id, $nt, &$links ) {
+		$user = RequestContext::getMain()->getUser();
+
+		if ( $id == 0 && $user->isAllowed( 'multilookup' ) ) {
+			$attribs = [
+				'href' => 'http://community.wikia.com/wiki/Special:MultiLookup?target=' . urlencode( $nt->getText() ),
+				'title' => wfMessage( 'multilookupselectuser' )->text(),
+			];
+
+			$links[] = Html::element( 'a', $attribs, wfMessage( 'multilookup' )->text() );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Update MultiLookup with IP activity each time an event is saved to Recent Changes
+	 * @param RecentChange $recentChange
+	 */
+	static public function onRecentChangeSave( RecentChange $recentChange ) {
+		$ip = $recentChange->getAttribute( 'rc_ip' );
+
+		if ( IP::isIPAddress( $ip ) ) {
+			global $wgCityId;
+
+			$task = ( new \Wikia\Tasks\Tasks\MultiLookupTask() )
+				->wikiId( $wgCityId );
+			$task->call( 'updateMultiLookup', $ip );
+			$task->queue();
+		}
```

Comment:
Does it make sense to log cases when `$ip` is not a proper IPv4/IPc6?

Reply:
If IP is not valid, we do not queue task here

Reply:
It can happen if some wiki has `wgPutIPinRC = false;`, unlikely, but possible.